tensorflow/core/kernels/image/draw_bounding_box_op.cc

```diff
@@ -165,37 +180,36 @@ class DrawBoundingBoxesOp : public OpKernel {
         CHECK_GE(max_box_col, 0);
 
         // Draw top line.
-        if (min_box_row >= 0) {
-          for (int64 j = min_box_col_clamp; j <= max_box_col_clamp; ++j)
-            for (int64 c = 0; c < depth; c++) {
-              canvas(b, min_box_row, j, c) =
-                  static_cast<T>(color_table[color_index][c]);
-            }
-        }
+        if (min_box_row >= 0)
```

Comment:
Isn't this condition no longer sufficient? Since the border can be arbitrarily thick and some might end up within-bounds.

I'm somewhat worried about out-of-bounds writes on the other lines too, e.g. the bottom line gets think enough to overflow the top of the image.

Reply:
I have updated the code to now have additional checks on both the inner limits and outer limits of the box (outer limits of the box are the limits passed by user in `boxes`. Inner limits are the limits within the box such that each line has passed thickness). Also  refactored for easier readability